lib/water_drop/message.rb

```diff
@@ -1,17 +1,19 @@
 module WaterDrop
   # Message class which encapsulate single Kafka message logic and its delivery
   class Message
-    attr_reader :topic, :message
+    attr_reader :topic, :message, :partition, :partition_key
 
     # @param topic [String, Symbol] a topic to which we want to send a message
     # @param message [Object] any object that can be serialized to a JSON string or
     #   that can be casted to a string
     # @return [WaterDrop::Message] WaterDrop message instance
     # @example Creating a new message
     #   WaterDrop::Message.new(topic, message)
-    def initialize(topic, message)
+    def initialize(topic, message, options = {})
```

Comment:
That makes sense. I was considering that but ended up creating accessors for the individual options for clarity. However, I'd be happy to have `message.options` instead and forward that to the producer. Let me know